### PR TITLE
Resolved clippy warnings

### PR DESCRIPTION
**Pull request by jD91mZM2**
Note: This does not fix cyclomatic complexity and #[inline(always)] warnings.
Neither does it fix [large enum variants](https://github.com/rust-lang-nursery/rust-clippy/wiki#large_enum_variant), because compatibility.
I want the last thing fixed some day though.

### CODE REVIEW

File: src/connection.rs [hunk 1]
```diff
@@ -75,10 +75,10 @@ impl Connection {
 			}
 		}
 
-		let (tx, rx) = mpsc::channel();
+		let (tx, mut rx) = mpsc::channel();
 		try!(::std::thread::Builder::new()
 			.name("Discord Keepalive".into())
-			.spawn(move || keepalive(heartbeat_interval, sender, rx)));
+			.spawn(move || keepalive(heartbeat_interval, sender, &mut rx)));
```
Comment (SpaceManiac) [inline src/connection.rs]: What's going on with these changes? What clippy lint is being claimed here?

File: src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 //! Client library for the [Discord](https://discordapp.com) API.
 //!
-//! The Discord API can be divided into three main components: the RESTful API
+//! The Discord API can be divided into three main components: the `RESTful` API
```
Comment (SpaceManiac) [inline src/lib.rs]: Incorrect

File: src/voice.rs [hunk 3]
```diff
@@ -370,18 +370,18 @@ enum Status {
...
 	};
 	::std::thread::Builder::new()
 		.name(name)
-		.spawn(move || voice_thread(rx))
+		.spawn(move || voice_thread(&mut rx))
 		.expect("Failed to start voice thread");
 }
 
-fn voice_thread(channel: mpsc::Receiver<Status>) {
+fn voice_thread(channel: &mut mpsc::Receiver<Status>) {
```
Comment (SpaceManiac) [inline src/voice.rs:384]: Bunch of stuff here that's also weird about the `mpsc::Receiver`?

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: src/connection.rs (hunk 1 above)
Comment (jD91mZM2) [reply src/connection.rs]: Clippy says it doesn't need to be a pass by value

File: src/lib.rs (hunk 2 above)
Comment (jD91mZM2) [reply src/lib.rs]: Ok. Will revert later

File: src/connection.rs (hunk 1 above)
Comment (SpaceManiac) [reply src/connection.rs]: I want them to be, exclude this (and the other `mpsc::Receiver` changes) and everything else should be good.

**Comment by jD91mZM2** [comment]
bump @SpaceManiac

**Comment by jD91mZM2** [comment]
nevermind sorry, didn't see you replied to the thingy

**Comment by SpaceManiac** [comment]
Fix #146

**Pull request closed without merging**